src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -1034,26 +1048,29 @@ protected void GenerateFindFirstChar()
 
                     if ((_anchors & RegexFCD.End) != 0)
                     {
-                        Label l1 = DefineLabel();
-                        Ldthisfld(s_runtextposField);
-                        Ldthisfld(s_runtextendField);
-                        Bge(l1);
-                        Ldthis();
-                        Ldthisfld(s_runtextendField);
-                        Stfld(s_runtextposField);
-                        MarkLabel(l1);
+                        if (minRequiredLength <= 0) // if it's > 0, we already output a more stringent check
```

Comment:
No need to change now, but in general I think your little comments showing the equivalent c# are worth having on every block of IL emitting like this.

Reply:
> I think your little comments showing the equivalent c# are worth having on every block of IL emitting like this

I agree.  I've added them everywhere I've added new code gen, but I've not gone back through all of the existing codegen to add them.  Worth doing at some point.

Reply:
Same for bunch of the blocks below